.github/workflows/macos.yml

```diff
@@ -29,3 +29,79 @@ jobs:
         env:
           MAX_WARNINGS: ${{ matrix.conf.max_warnings }}
         run:  python3 ./scripts/count-warnings.py build.log
+
+  build_macos_release:
+    name: Release build
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install C++ compiler and libraries
+        run:  brew install librsvg $(./scripts/list-build-dependencies.sh -m brew -c clang)
+      - name: Log environment
+        run:  ./scripts/log-env.sh
+      - name: Inject version string
+        run: |
+          set -x
+          export VERSION=$(git describe --abbrev=5)
+          sed -i -e "s/AC_INIT(dosbox,git)/AC_INIT(dosbox,$VERSION)/" configure.ac
+          echo ::set-env name=VERSION::$VERSION
+      - name: Build
+        env:
+          FLAGS: -O3 -DNDEBUG -pipe
```

Comment:
The silver lining with exclusively supporting macOS 10.15 Catalina (and not prior versions), means we can assume lower bound in processor support based on the oldest machine that Catalina will run on, and that is the Ivy Bridge-based architecture.

So we can safely add `-march=ivybridge` to this build string, which means GCC can make use of the mmx, sse, sse2, pclmulqdq, ssse3, sse4_1, sse4_2, and avx instructions.  

---
Which Macs can run macOS 10.15 Catalina?
- MacBook (2015 or newer) - i5 Broadwell
- MacBook Air (2012 or newer) - i5 Ivy Bridge
- MacBook Pro (2012 or newer) - i5 Ivy Bridge
- Mac mini (2012 or newer) - i5 Ivy Bridge
- iMac (2012 or newer) - i5 Ivy Bridge
- iMac Pro (2017 or newer) - Xeon W-2150B (Skylake architecture)
- Mac Pro (2013 or newer) - Xeon E5-1620v2 (Ivy Bridge EP architecture)

Reply:
What about hackintoshes or virtual machines? We can't be sure Catalina won't be running on any particular hardware configuration.

Reply:
My guess is that Catalina itself is built with those architecture instructions, similar to how Windows 10 can only run on hardware going so far back because they wanted to also use accelerated instructions.

Reply:
After a cursory internet search, it appears folks have been successfully running Catalina on at least Sandy Bridge.

Reply:
Good call @JoshuaFern; Hackintosh says:
> Catalina requires a processor with the SSE4.2 instruction.
Ref: https://www.hackintoshzone.com/files/file/1090-niresh-catalina/

SSE 4.2 first arrived with the Nahelem architecture; so it looks like Catalina does go back slightly further in time than Ivy Bridge.  Updating my original comment to replace Ivy Bridge.

Reply:
This is an excellent suggestion - thanks @krcroft, @JoshuaFern; implemented in 72b737fc94149d6a2f969bb7b9afad33125daeb2.

Now I wonder: should we enable Nehalem also for our Linux snapshot builds? (that would be for a separate PR though).

Reply:
It's a reasonable cutoff; given the processors are now 12 years old.  My guess is this would cover 99% of dosbox-staging snapshot users (maybe even more).  The other 1% will likely be building their own binaries, especially if they're running on really old hardware (like myself on that old big-endian PowerPC machine).